docs/tutorials/getting-started-with-csharp/working-with-linq.md

```diff
@@ -0,0 +1,510 @@
+
+Don't misinterpret this example by thinking that all queries should run eagerly. This is a particularly
+pathological case. In reality, some algorithms run much better using eager evaluation, and others run much
+better using lazy evaluation. LINQ enables both. Measure, and pick the best choice.
+
+# Improving Readability
+
+The code you've written for this sample is an example of creating a simple prototype that does the
+job. You've leveraged *anonymous types* for the cards, and each card is represented by strings. 
+
+Let's clean those up somewhat. Start with the suits. This is a perfect time to use an `enum`:
+
+```cs
+public enum Suit
+{
+    Clubs,
+    Diamonds,
+    Hearts,
+    Spades
+}
+```
+
+The `Suits()` method also changes type and implementation:
+
+```cs
+static IEnumerable<Suit> Suits()
+{
+    yield return Suit.Clubs;
+    yield return Suit.Diamonds;
+    yield return Suit.Hearts;
+    yield return Suit.Spades;
+}
+```
+
+Next, do the same change with the Rank of the cards:
+
+```cs
+public enum Rank
+{
+    Two,
+    Three,
+    Four,
+    Five,
+    Six,
+    Seven,
+    Eight,
+    Nine,
+    Ten,
+    Jack,
+    Queen,
+    King,
+    Ace
+}
+```
+
+And the method that generates them:
+
+```cs
+static IEnumerable<Rank> Values()
+{
+    yield return Rank.Two;
+    yield return Rank.Three;
+    yield return Rank.Four;
+    yield return Rank.Five;
+    yield return Rank.Six;
+    yield return Rank.Seven;
+    yield return Rank.Eight;
+    yield return Rank.Nine;
+    yield return Rank.Ten;
+    yield return Rank.Jack;
+    yield return Rank.Queen;
+    yield return Rank.King;
+    yield return Rank.Ace;
+}
+```
+
+As one final cleanup, let's make a type to represent the card, instead of 
+relying on an anonymous types. Anonymous types are great for lightweight,
+local types, but in this example, the playing card is one of the main
+concepts. It should be a concrete type.
+
+```cs
+public class PlayingCard
+{
+    public Suit CardSuit { get; }
+    public Rank CardRank { get; }
+    
+    public PlayingCard(Suit s, Rank r)
+    {
+        CardSuit = s;
+        CardRank = r;
+    }
+    
+    public override string ToString()
+    {
+        return $"{CardRank} of {CardSuit}";
+    }
+}
+```
+
+This type uses *auto-implemented read-only properties* which are set
+in the constructor, and then cannot be modified. It also makes use of
+the new *string interpolation* feature that makes it easier to format
+string output.
+
+Update the query that generates the starting deck to use the new type:
+
+```cs
+var startingDeck = (from s in Suits().LogQuery("Suit Generation")
+                    from r in Ranks().LogQuery("Value Generation")
+                    select new PlayingCard(s, r))
+                    .LogQuery("Starting Deck")
+                    .ToArray();
+```
+
+Compile and run again. The output is a little cleaner, and the code is a bit
+more clear and can be extended more easily.
```

Comment:
This section seems to stray from the main focus of the tutorial since it doesn't really teach more about LINQ.  Maybe it could be tweaked somehow to emphasize some advantage or difference in the query expressions that might result from the code improvements?